app/controllers/UploadSupportingMaterialMultipleController.scala

```diff
@@ -62,55 +63,78 @@ class UploadSupportingMaterialMultipleController @Inject()(
     numberOfFiles >= 10
   }
 
-  def addFile(file: FileAttachment, userAnswers: UserAnswers): UserAnswers = {
+  def upsertFile(file: FileAttachment, userAnswers: UserAnswers): UserAnswers = {
     val updatedFiles = userAnswers
       .get(UploadSupportingMaterialMultiplePage)
-      .map(files => files :+ file)
-      .getOrElse(Seq(file))
+      .map { files =>
+        val index = files.indexWhere(_.id == file.id)
+        if (index >= 0) {
+          files.take(index) ++ Seq(file) ++ files.drop(index + 1)
+        } else {
+          files :+ file
+        }
+      }.getOrElse(Seq(file))
```

Comment:
It might make sense to filter out files where `uploaded = false` here to ensure this is kept up to date